src/ssl.c

```diff
@@ -40129,17 +42400,39 @@ void* wolfSSL_GetDhAgreeCtx(WOLFSSL* ssl)
 
 #ifdef OPENSSL_EXTRA
 
-#ifndef NO_WOLFSSL_STUB
     int wolfSSL_X509_check_private_key(WOLFSSL_X509 *x509, WOLFSSL_EVP_PKEY *key)
     {
-        (void) x509;
-        (void) key;
+        DecodedCert dc;
+        byte* der;
+        int derSz;
+        int ret;
+
         WOLFSSL_ENTER("wolfSSL_X509_check_private_key");
-        WOLFSSL_STUB("X509_check_private_key");
 
-        return WOLFSSL_SUCCESS;
+        if (!x509 || !key) {
+            WOLFSSL_MSG("Bad parameter");
+            return WOLFSSL_FAILURE;
+        }
+
+        der = (byte*)wolfSSL_X509_get_der(x509, &derSz);
```

Comment:
Can't we get the public key out of the WOLFSSL_X509 structure?

Reply:
We probably could but we don't have any API that takes in just a public key without a certificate for verification.

Reply:
The function is checking that the public key matches the private key.
OpenSSL doesn't validate the certificate, it just extracts the public key and compares to the private key.

Reply:
Refactored wc_CheckPrivateKey to accept buffers and added wc_CheckPrivateKeyCert if we want to pass in DecodedCert instead.